src/app.js

```diff
@@ -0,0 +1,191 @@
+  listCountElement.innerText = `${incompleteTaskCount} ${taskString} remaining`;
+};
+
+const renderList = () => {
+  lists.forEach((list) => {
+    const listElement = document.createElement("li");
+    listElement.dataset.listId = list.id;
+    listElement.classList.add("list-name");
+    listElement.innerText = list.topic;
+    if (list.id === selectedListId) {
+      listElement.classList.add("active-list");
+    }
+    listsContainer.appendChild(listElement);
+  });
+};
+
+const saveToLocal = () => {
+  localStorage.setItem(LOCAL_STORAGE_LIST_KEY, JSON.stringify(lists));
+  localStorage.setItem(LOCAL_STORAGE_SELECTED_LIST_ID_KEY, selectedListId);
+};
+
+const saveAndRender = () => {
+  saveToLocal();
+  render();
+};
+
+newListForm.addEventListener("submit", (e) => {
+  e.preventDefault();
+  const listName = newListInput.value;
+  if (listName == null || listName === "") return;
+  const list = new createList(listName);
+  newListInput.value = null;
+  lists.push(list);
+  saveAndRender();
+});
+
+newTaskForm.addEventListener("submit", (e) => {
+  e.preventDefault();
+  const titleName = newListTitle.value;
+  const descName = newListDesc.value;
+  const dateName = newListDate.value;
+  const prioName = newPrioInput.options[newPrioInput.selectedIndex].text;
+  if (
+    titleName == null ||
+    titleName === "" ||
+    descName == null ||
+    descName === "" ||
+    dateName == null ||
+    dateName === "" ||
+    prioName == null ||
+    prioName === ""
+  )
+    // eslint-disable-next-line nonblock-statement-body-position
+    return;
+  const task = new createTask(titleName, descName, dateName, prioName);
+  newListTitle.value = null;
+  newListDesc.value = null;
+  newListDate.value = null;
+  newPrioInput.options[newPrioInput.selectedIndex].value = null;
+  const selectedList = lists.find((list) => list.id === selectedListId);
+  selectedList.tasks.push(task);
+  saveAndRender();
+});
+
+listsContainer.addEventListener("click", (e) => {
+  if (e.target.tagName.toLowerCase() === "li") {
+    selectedListId = e.target.dataset.listId;
+    saveAndRender();
+  }
+});
+
+taskContainer.addEventListener("click", (e) => {
+  if (e.target.tagName.toLowerCase() === "input") {
+    const selectedList = lists.find((list) => list.id === selectedListId);
+    const selectedTask = selectedList.tasks.find(
+      (task) => task.id === e.target.id
+    );
+    selectedTask.complete = e.target.checked;
+    saveToLocal();
+    renderTaskCount(selectedList);
+  }
+});
+
+clearCompleteTasksButton.addEventListener("click", (e) => {
+  const selectedList = lists.find((list) => list.id === selectedListId);
+  selectedList.tasks = selectedList.tasks.filter((task) => !task.complete);
+  saveAndRender();
+});
+
+deleteListButton.addEventListener("click", (e) => {
+  lists = lists.filter((list) => list.id !== selectedListId);
+  selectedListId = null;
+  saveAndRender();
+});
+
+function createList(topic) {
```

Comment:
- [ ] It is better to use the ES6 arrow functions instead of using the function keywords. Make the changes where necessary for all your functions